x-pack/plugins/ml/public/explorer/explorer_controller.js

```diff
@@ -70,279 +54,93 @@ uiRoutes
-  });
-
-  const dragSelect = new DragSelect({
-    selectables: document.getElementsByClassName('sl-cell'),
-    callback(elements) {
-      if (elements.length > 1 && !ALLOW_CELL_RANGE_SELECTION) {
-        elements = [elements[0]];
-      }
-
-      if (elements.length > 0) {
-        mlExplorerDashboardService.dragSelect.changed({
-          action: DRAG_SELECT_ACTION.NEW_SELECTION,
-          elements
-        });
-      }
-
-      disableDragSelectOnMouseLeave = true;
-    },
-    onDragStart() {
-      if (ALLOW_CELL_RANGE_SELECTION) {
-        mlExplorerDashboardService.dragSelect.changed({
-          action: DRAG_SELECT_ACTION.DRAG_START
-        });
-        disableDragSelectOnMouseLeave = false;
-      }
-    },
-    onElementSelect() {
-      if (ALLOW_CELL_RANGE_SELECTION) {
-        mlExplorerDashboardService.dragSelect.changed({
-          action: DRAG_SELECT_ACTION.ELEMENT_SELECT
-        });
-      }
-    }
-  });
-
-  $scope.selectedJobs = null;
-
-  $scope.getSelectedJobIds = function () {
-    const selectedJobs = _.filter($scope.jobs, job => job.selected);
-    return _.map(selectedJobs, job => job.id);
-  };
-
-  $scope.viewBySwimlaneOptions = [];
-  $scope.viewBySwimlaneData = getDefaultViewBySwimlaneData();
-  $scope.viewBySwimlaneDataLoading = false;
-
-  $scope.initializeVis = function () {
-    // Initialize the AppState in which to store filters.
-    const stateDefaults = {
-      filters: [],
-      mlExplorerSwimlane: {}
-    };
-    $scope.appState = new AppState(stateDefaults);
-    $scope.jobs = [];
-
-    // Load the job info needed by the dashboard, then do the first load.
-    // Calling loadJobs() ensures the full datafeed config is available for building the charts.
-    mlJobService.loadJobs().then((resp) => {
-      if (resp.jobs.length > 0) {
-        $scope.jobs = createJobs(resp.jobs);
-
-        // Select any jobs set in the global state (i.e. passed in the URL).
-        const selectedJobIds = mlJobSelectService.getSelectedJobIds(true);
-        $scope.setSelectedJobs(selectedJobIds, true);
-      } else {
-        $scope.loading = false;
-      }
-
-      $scope.$applyAsync();
-    }).catch((resp) => {
-      console.log('Explorer - error getting job info from elasticsearch:', resp);
-    });
-
-    mlExplorerDashboardService.init();
-  };
+  mlExplorerDashboardService.init();
 
-  // create new job objects based on standard job config objects
-  // new job objects just contain job id, bucket span in seconds and a selected flag.
-  function createJobs(jobs) {
-    return jobs.map(job => {
-      const bucketSpan = parseInterval(job.analysis_config.bucket_span);
-      return { id: job.job_id, selected: false, bucketSpanSeconds: bucketSpan.asSeconds() };
-    });
-  }
+  // Load the job info needed by the dashboard, then do the first load.
+  // Calling loadJobs() ensures the full datafeed config is available for building the charts.
+  mlJobService.loadJobs().then(async (resp) => {
```

Comment:
i'm not sure this function needs to be `async`

Reply:
Done in 65b1835d1b.